### PR TITLE
tests: allow pilot suite to run with more than one cluster

### PR DESCRIPTION
**Pull request by stevenctl**
Split the first few commits out from https://github.com/istio/istio/pull/24549 as a first step in breaking up that PR

This PR shouldn't change the behavior of any tests. It just allows us to run `integ-pilot-multicluster-tests_istio` without the whole suite being skipped and incrementally convert pilot tests in separate PRs. 

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: tests/integration/pilot/main_test.go [hunk 1]
```diff
@@ -18,18 +18,17 @@ import (
...
-	"istio.io/istio/pkg/test/framework/components/echo"
-	"istio.io/istio/pkg/test/framework/components/namespace"
-
 	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
 	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
 	"istio.io/istio/pkg/test/framework/components/pilot"
-	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/tests/integration/multicluster"
```
Comment (howardjohn) [inline tests/integration/pilot/main_test.go:26]: I wonder if this should be in pkg/test/framework instead? may not be a big concern though

**Comment by howardjohn** [comment]
LGTM, will let nate have a chance to look at it

File: tests/integration/pilot/main_test.go (hunk 1 above)
Comment (nmittler) [reply tests/integration/pilot/main_test.go:26]: +1 mind making that move?

**Comment by stevenctl** [comment]
@nmittler should be good to go

[integ-pilot-multicluster-tests_istio](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25432/integ-pilot-multicluster-tests_istio/37) sets up the suite and skips unimplemented tests

When this and https://github.com/istio/istio/pull/25139 land we can start pulling in test conversions (splitting up https://github.com/istio/istio/pull/24549 https://github.com/istio/istio/pull/25055)

File: pkg/test/framework/components/pilot/pilot.go [hunk 2]
```diff
@@ -69,6 +65,32 @@ func NewOrFail(t test.Failer, c resource.Context, config Config) Instance {
 	return i
 }
 
+// Setup returns a test framework setup function that will populate the passed in slice with
+// a pilot Instance pointing to the control plane cluster for the cluster at each index.
+func Setup(pilots *[]Instance, baseConfig Config) resource.SetupFn {
```
[the remark this answers is not in the document]
Comment (stevenctl) [reply pkg/test/framework/components/pilot/pilot.go:70]: https://github.com/istio/istio/issues/25577 I'll work on that and clean this up when it's done.

File: pkg/test/framework/components/pilot/kube.go [hunk 3]
```diff
@@ -45,12 +45,12 @@ func newKube(ctx resource.Context, cfg Config) (Instance, error) {
 	}
 	c.id = ctx.TrackResource(c)
 
-	// TODO: This should be obtained from an Istio deployment.
-	icfg, err := istio.DefaultConfig(ctx)
+	var i istio.Instance
```
[the remark this answers is not in the document]
Comment (stevenctl) [reply pkg/test/framework/components/pilot/kube.go]: done

**Pull request merged**